runtime/bindings/python/hal.cc

```diff
@@ -917,33 +885,25 @@ void SetupHalBindings(nanobind::module_ m) {
             params.type = memory_type;
             params.usage = allowed_usage;
             iree_hal_buffer_t* buffer = nullptr;
-            iree_const_byte_span_t empty_initial_data{nullptr, 0};
-            CheckApiStatus(iree_hal_allocator_allocate_buffer(
-                               self.raw_ptr(), params, allocation_size,
-                               empty_initial_data, &buffer),
-                           "could not allocate buffer");
+            CheckApiStatus(
+                iree_hal_allocator_allocate_buffer(self.raw_ptr(), params,
+                                                   allocation_size, &buffer),
+                "could not allocate buffer");
             return HalBuffer::StealFromRawPtr(buffer);
           },
           py::arg("memory_type"), py::arg("allowed_usage"),
           py::arg("allocation_size"), py::keep_alive<0, 1>(),
           "Allocates a new buffer with requested characteristics (does not "
           "initialize with specific data).")
       .def("allocate_buffer_copy", &HalAllocator::AllocateBufferCopy,
-           py::arg("memory_type"), py::arg("allowed_usage"), py::arg("buffer"),
-           py::arg("element_type") = py::none(), py::keep_alive<0, 1>(),
+           py::arg("memory_type"), py::arg("allowed_usage"), py::arg("device"),
+           py::arg("buffer"), py::arg("element_type") = py::none(),
+           py::keep_alive<0, 1>(),
            "Allocates a new buffer and initializes it from a Python buffer "
            "object. If an element type is specified, wraps in a BufferView "
            "matching the characteristics of the Python buffer. The format is "
            "requested as ND/C-Contiguous, which may incur copies if not "
-           "already in that format.")
-      .def("allocate_host_staging_buffer_copy",
```

Comment:
ah I couldn't find it on github/sourcegraph code search so assumed it was dead - I can add it back but the implementation likely needs to change - maybe you can use files now anyway

Reply:
Added it back - `iree_hal_allocator_import_buffer` is getting better and that's one route to avoid copies, or if your tensor types are async you could schedule an async copy here with the new `iree_hal_device_queue_copy` helper (which creates a copy command buffer and submits it for you)